### ISSUE TITLE
Can't reset header select filter in 4.2

### ISSUE THREAD

**Issue by GuzziGuy** [olifolkerd/tabulator#2160]
Using header select filter with select options, but it seems impossible to unselect a filter once chosen, to get back to all data.

This is visible in the demo for header filters:
http://tabulator.info/examples/4.2?#filter-header

Note the 'Favourite color' column - once you select a colour, you can't undo it and go back to viewing all.

For reference, the source displayed on the docs seems incorrect, it shows:

`{title:"Favourite Color", field:"col", headerFilter:"input"},`

But the actual code is this (which is exactly what I'm doing):

`{title:"Favourite Color", field:"col", editor:"input", headerFilter:"select", headerFilterParams:{values:true}},`

I did then think to DIY and build a unique list (+blank/all) by using a function for headerFilterParams but as far as I can tell, this function is only called once, and it receives no parameters, so the table object is not accessible to get the data from.

I'm likely missing something obvious! But if not, it would be a handy option to add to this...

**Comment by tentus** [comment]
Probably would be solved by https://github.com/olifolkerd/tabulator/pull/1881

**Comment by GuzziGuy** [comment]
#1881 would fix this - as I understand it makes the entry field essentially editable, so you could erase whatever has been selected in the menu? This should work, but I think that adding a 'Show all' entry to the select would be more expected UI?

For now I've worked around it with some clunky jquery, so it's not a show-stopper.

**Comment by olifolkerd** [comment]
I will look at pulling this in to the next release.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
#1881 has been merged in for tomorrows release.

Cheers

Oli :)

**Comment by gpremkumar** [comment]
Hello Oli,
 Just a quick bit related to above question.
Do we have the option added to reset Header filter. I added **headerFilterParams:{values:true}** option in column Definition. I couldn't see 'x' mark to reset filter.

I added this option in column definition.
{
title: "Type", field: "type", editor: 'select',  
headerFilterPlaceholder: "filter", 
headerFilter: true,
headerFilterParams:{values:true}
}

Tabulator version used 4.7

Cheers,
Prem
